docs/development/new_cp_support.md

```diff
@@ -36,35 +47,70 @@ Steps to be followed while implementing a new (hyperscale) provider are mentione
 1. Now commit your changes and push it upstream.
     ```bash
     git add -A
-    git commit
+    git commit -m "Renamed SampleProvider"
     git push origin master
     ```
 
 ### Code changes required
 
-1. Update the `pkg/{provider-name}/apis/provider-spec.go` specification file to reflect the structure of the objects exchanged between the machine-controller-manager and the driver. It typically contains the machine details and secrets (if required). Follow the sample spec provided already in the file. A sample provider specification can be found [here](https://github.com/prashanth26/machine-controller-manager-provider-gcp/blob/master/pkg/gcp/apis/provider-spec.go). 
-    ```bash
-    vim pkg/{provider-name}/apis/provider-spec.go
-    ```
-1. Fill in the create/list/delete methods `pkg/{provider-name}/machineserver.go` to create/list/delete VMs on your cloud provider. A sample provider implementation for these methods can be found [here](https://github.com/prashanth26/machine-controller-manager-provider-gcp/blob/master/pkg/gcp/machineserver.go).
-    ```bash
-    vim pkg/{provider-name}/machineserver.go
-    ```
+1. Update the `pkg/{provider-name}/apis/provider-spec.go` specification file to reflect the structure of the objects exchanged between the machine-controller-manager (AKA cmi-client) and the driver (AKA cmi-server). It typically contains the machine details and secrets (if required). Follow the sample spec provided already in the file. A sample provider specification can be found [here](https://github.com/prashanth26/machine-controller-manager-provider-gcp/blob/master/pkg/gcp/apis/provider-spec.go).
+1. Fill in the methods described at `pkg/{provider-name}/machineserver.go` to manage VMs on your cloud provider.
+    - To test the basic working of your driver, implementing just the create & delete would suffice.
+    - The request and response parameters for each of the methods to be implemented are well documented as comments and sample codes at `pkg/{provider-name}/machineserver.go`.
+    - Once, create and delete methods behave as expected, you can fill in the rest of the methods.
```

Comment:
I hoped we would have expected a user to implement GetMachine as well, seems something very basic to be implemented for features like safety-controller and in case we implement GetMachine before every CreateMachine call on mcm. Leaving the decision to you.

Reply:
I have made the suggested changes and mandated basic methods - https://github.com/gardener/machine-controller-manager/pull/221/files#diff-d01da958f03686a1e51543cad6dcd458R58